Vignettes/An Introduction to ‘matchFeats’Rmd.Rmd

```diff
@@ -0,0 +1,217 @@
+
+The optional argument \code{control} can be specified with these fields: 
+\code{maxit}, maximum number of EM iterations (default=1e4); 
+\code{eps}, relative tolerance for EM convergence (default=1e-8), 
+the EM algorithm stops if the relative increase in log-likelihood between two iterations is less than this tolerance; \code{verbose}, set to TRUE to display
+ algorithm progress (default=FALSE). 
+
+#### Example
+
+#Issue(need to solve)
+``` 
+## Generate data
+  m <- 3
+  n <- 10
+  p <- 5
+  mu <- matrix(rnorm(p*m),p,m)
+  sigma <- 0.1
+  x <- array(mu + rnorm(p*m*n,sigma), c(p,m,n))
+```
+
+Import matrixes, then:
+```{r}
+## Match all feature vectors
+  result <- match.kmeans(x)
+
+## Display results 
+  result$cost # cost function
+  xmatched <- array(dim=dim(x)) 
+  
+## re-arranged (matched) feature vectors
+  for (i in 1:n){
+	  xmatched[,,i] <- x[,result$sigma[,i],i]}
+```
+
+### Template-Based Method(`match.template`)
+
+#### Details
+
+Given a sample of \eqn{n} statistical units, each having \eqn{m} possibly mislabeled feature vectors, the one-to-one matching problem is to find a set of \eqn{n} label permutations that produce the best match of feature vectors across units. The objective function to minimize is the sum of squared (Euclidean) distances between all feature vectors having the same (new) label. This amounts to minimizing the sum of the within-label variances.  
+	
+The template-based method consists in relabeling successively each sample unit to best match a template matrix of feature vectors. This method is very fast but its optimization performance is only as good as the template. A template that is typical or representative of the collected data. 
+
+If \code{x} is a matrix, the rows should be sorted by increasing unit label and  \code{unit} should be a nondecreasing sequence of integers, for example \eqn{(1,...,1,2,...,2,...,n,...,n)} with each integer \eqn{1,...,n} replicated \eqn{m} times. 
+
+The argument \code{w} can be specified as a vector of positive numbers (will be recycled to length \eqn{p} if needed) or as a positive definite matrix of size \eqn{(p,p)}.
+
+#### Example
+
+Import matrixes, then:
+```{r}
+## Match all feature vectors with first case as template
+result <- match.template(x,1)
+
+## Display results 
+result$cost # cost function
+xmatched <- array(dim=dim(x)) 
+
+# re-arranged (matched) feature vectors
+for (i in 1:n){
+	xmatched[,,i] <- x[,result$sigma[,i],i]
+}
+
+```
+
+## Handwriting example project
+
+The data comes from UCI - Optical Recognition of Handwritten Digits Data Set which used to test matchFeats packages. 
+There are 3823 training data and 1797 testing data in this handwritten digits data set example projec.  Each data set has 
+```{r}
+train_url <- "https://archive.ics.uci.edu/ml/machine-learning-databases/optdigits/optdigits.tra"
+train <- read.table(train_url,sep = ",")
+str(train)
+```
+
+## Shiny App (Future Function, if it necessary)
```

Comment:
Delete or comment out this part. Only include in the vignette features that currently exist, not future ones.